src/java.base/share/classes/java/security/SecureRandom.java

```diff
@@ -189,6 +190,38 @@
     // Seed Generator
     private static volatile SecureRandom seedGenerator;
 
+    /*
+     * A negative value indicating a period larger than 2**(2**16).
+     */
+    private static final BigInteger PERIOD = RandomGenerator.HUGE_PERIOD;
+
+    /*
+     * Number of bits used to maintain state of seed. A large value indicating
+     * stateBits is unknown.
+     */
+    private static final int STATE_BITS = Integer.MAX_VALUE;
+
+    /*
+     * The equidistribution of the algorithm. A large value indicating
+     * stateBits is unknown.
+     */
+    private static final int EQUIDISTRIBUTION = Integer.MAX_VALUE;
+
+    /*
+     * RandomGenerator properties.
+     */
+    static Map<RandomGeneratorProperty, Object> getProperties() {
+        return Map.ofEntries(
+                Map.entry(RandomGeneratorProperty.NAME, "SecureRandom"),
+                Map.entry(RandomGeneratorProperty.GROUP, "Legacy"),
+                Map.entry(RandomGeneratorProperty.PERIOD, PERIOD),
+                Map.entry(RandomGeneratorProperty.STATE_BITS, STATE_BITS),
+                Map.entry(RandomGeneratorProperty.EQUIDISTRIBUTION, EQUIDISTRIBUTION),
+                Map.entry(RandomGeneratorProperty.IS_STOCHASTIC, true),
+                Map.entry(RandomGeneratorProperty.IS_HARDWARE, false)
+        );
+    }
```

Comment:
Using Map.of() instead of Map.ofEntries() should simplify the code

Reply:
I had assumed  Map.ofEntries() was more efficient but it seems it in turn uses MapN as well. Will change these cases.